python/lsst/sims/catalogs/generation/db/dbConnection.py

```diff
@@ -84,20 +84,45 @@ def _postprocess_results(self, chunk):
 
 class DBObject(object):
 
-    def __init__(self, address=None, verbose=False):
-        self.verbose=verbose
+    def __init__(self, driver=None, host=None, port=None, database=None, verbose=False):
 
-        self.dtype = None 
-        #this is a cache for the query, so that any one query does not have to guess dtype multiple times
+        #Override the class variables if supplied arguments
+        if driver is not None:
+            self.driver = driver
+        if host is not None:
+            self.host = host
+        if port is not None:
+            self.port = port
+        if database is not None:
+            self.database = database
+
+        if self.driver == 'sqlite':
+            #When passed sqlite database, override default host/port
+            self.host = None
+            self.port = None
 
-        if address is not None:
-            self.dbAddress = address
+        self.verbose=verbose
+        self.dtype = None
+        #this is a cache for the query, so that any one query does not have to guess dtype multiple times
 
         self._connect_to_engine()
 
     def _connect_to_engine(self):
         """create and connect to a database engine"""
-        self.engine = create_engine(self.dbAddress, echo=self.verbose)
+
+        #DbAuth will not look up hosts that are None, '' or 0
+        if self.host:
```

Comment:
I added a validation method, a more informative error message if connection fails and username/password kwargs which print a warning if you use them and throw an exception if the username is our shared username.
I'm still conflicted on the username/password kwargs: 
 * We don't want to use them. 
 * They are simple to add in the future, yet difficult to remove once they become used in client code.
 *  Good APIs have one way to do things. 
 *  Users ignore warnings.  Presence of username/password kwargs look like an invitation to use them. 
 *  Passwords might creep into example scripts and demo ipython notebook repositories.